### DISCUSSION TITLE
Optional fsspec filesystems

### DISCUSSION THREAD

**Discussion by legout** [delta-io/delta-rs#2947] [Q&A]
# Description

Make Python filesystem spec (fsspec) optional available in delta-rs.

**Use Case**

One could easily cache metadata files/delta log using fsspecs `SimpleCacheFileSystem` with s3fs, gcsfs or adlfs. This might improve read performance.

**Comment by ion-elgreco** [comment]
We won't support fsspec filesystems since we use object store. You can create a custom object store which does caching and make a PR to add this store that can be enabled with a storage option
